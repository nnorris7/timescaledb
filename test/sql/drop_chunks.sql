\o /dev/null
\ir include/create_single_db.sql
\o

CREATE TABLE PUBLIC.drop_chunk_test1(time bigint, temp float8, device_id text);
CREATE TABLE PUBLIC.drop_chunk_test2(time bigint, temp float8, device_id text);
CREATE INDEX ON drop_chunk_test1(time DESC);
SELECT create_hypertable('public.drop_chunk_test1', 'time', chunk_time_interval => 1, create_default_indexes=>false);
SELECT create_hypertable('public.drop_chunk_test2', 'time', chunk_time_interval => 1, create_default_indexes=>false);

SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');

\dt "_timescaledb_internal".*

SELECT  _timescaledb_internal.get_partition_for_key('dev1');
SELECT  _timescaledb_internal.get_partition_for_key('dev7');

INSERT INTO PUBLIC.drop_chunk_test1 VALUES(1, 1.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(2, 2.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(3, 3.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(4, 4.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(5, 5.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(6, 6.0, 'dev7');

INSERT INTO PUBLIC.drop_chunk_test2 VALUES(1, 1.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(2, 2.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(3, 3.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(4, 4.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(5, 5.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(6, 6.0, 'dev7');

SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');
\dt "_timescaledb_internal".*

SELECT _timescaledb_internal.drop_chunks_older_than(2);

SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');

\dt "_timescaledb_internal".*

SELECT _timescaledb_internal.drop_chunks_older_than(3, 'drop_chunk_test1');

SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');

\dt "_timescaledb_internal".*
