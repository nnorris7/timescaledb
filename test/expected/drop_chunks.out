\o /dev/null
\ir include/create_single_db.sql
SET client_min_messages = WARNING;
DROP DATABASE IF EXISTS single;
SET client_min_messages = NOTICE;
CREATE DATABASE single;
\c single
CREATE EXTENSION IF NOT EXISTS timescaledb;
\o
CREATE TABLE PUBLIC.drop_chunk_test1(time bigint, temp float8, device_id text);
CREATE TABLE PUBLIC.drop_chunk_test2(time bigint, temp float8, device_id text);
CREATE INDEX ON drop_chunk_test1(time DESC);
SELECT create_hypertable('public.drop_chunk_test1', 'time', chunk_time_interval => 1, create_default_indexes=>false);
 create_hypertable 
-------------------
 
(1 row)

SELECT create_hypertable('public.drop_chunk_test2', 'time', chunk_time_interval => 1, create_default_indexes=>false);
 create_hypertable 
-------------------
 
(1 row)

SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');
 chunk_id | hypertable_id | chunk_schema | chunk_table | range_start | range_end 
----------+---------------+--------------+-------------+-------------+-----------
(0 rows)

\dt "_timescaledb_internal".*
      List of relations
 Schema | Name | Type | Owner 
--------+------+------+-------
(0 rows)

SELECT  _timescaledb_internal.get_partition_for_key('dev1');
 get_partition_for_key 
-----------------------
            1129986420
(1 row)

SELECT  _timescaledb_internal.get_partition_for_key('dev7');
 get_partition_for_key 
-----------------------
             449729092
(1 row)

INSERT INTO PUBLIC.drop_chunk_test1 VALUES(1, 1.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(2, 2.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(3, 3.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(4, 4.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(5, 5.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(6, 6.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(1, 1.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(2, 2.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(3, 3.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(4, 4.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(5, 5.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(6, 6.0, 'dev7');
SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');
 chunk_id | hypertable_id |     chunk_schema      |    chunk_table    | range_start | range_end 
----------+---------------+-----------------------+-------------------+-------------+-----------
        1 |             1 | _timescaledb_internal | _hyper_1_1_chunk  |           1 |         2
        2 |             1 | _timescaledb_internal | _hyper_1_2_chunk  |           2 |         3
        3 |             1 | _timescaledb_internal | _hyper_1_3_chunk  |           3 |         4
        4 |             1 | _timescaledb_internal | _hyper_1_4_chunk  |           4 |         5
        5 |             1 | _timescaledb_internal | _hyper_1_5_chunk  |           5 |         6
        6 |             1 | _timescaledb_internal | _hyper_1_6_chunk  |           6 |         7
        7 |             2 | _timescaledb_internal | _hyper_2_7_chunk  |           1 |         2
        8 |             2 | _timescaledb_internal | _hyper_2_8_chunk  |           2 |         3
        9 |             2 | _timescaledb_internal | _hyper_2_9_chunk  |           3 |         4
       10 |             2 | _timescaledb_internal | _hyper_2_10_chunk |           4 |         5
       11 |             2 | _timescaledb_internal | _hyper_2_11_chunk |           5 |         6
       12 |             2 | _timescaledb_internal | _hyper_2_12_chunk |           6 |         7
(12 rows)

\dt "_timescaledb_internal".*
                      List of relations
        Schema         |       Name        | Type  |  Owner   
-----------------------+-------------------+-------+----------
 _timescaledb_internal | _hyper_1_1_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_2_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_3_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_4_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_5_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_6_chunk  | table | postgres
 _timescaledb_internal | _hyper_2_10_chunk | table | postgres
 _timescaledb_internal | _hyper_2_11_chunk | table | postgres
 _timescaledb_internal | _hyper_2_12_chunk | table | postgres
 _timescaledb_internal | _hyper_2_7_chunk  | table | postgres
 _timescaledb_internal | _hyper_2_8_chunk  | table | postgres
 _timescaledb_internal | _hyper_2_9_chunk  | table | postgres
(12 rows)

SELECT _timescaledb_internal.drop_chunks_older_than(2);
NOTICE:  index "1-drop_chunk_test1_time_idx" does not exist, skipping
 drop_chunks_older_than 
------------------------
 
(1 row)

SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');
 chunk_id | hypertable_id |     chunk_schema      |    chunk_table    | range_start | range_end 
----------+---------------+-----------------------+-------------------+-------------+-----------
        2 |             1 | _timescaledb_internal | _hyper_1_2_chunk  |           2 |         3
        3 |             1 | _timescaledb_internal | _hyper_1_3_chunk  |           3 |         4
        4 |             1 | _timescaledb_internal | _hyper_1_4_chunk  |           4 |         5
        5 |             1 | _timescaledb_internal | _hyper_1_5_chunk  |           5 |         6
        6 |             1 | _timescaledb_internal | _hyper_1_6_chunk  |           6 |         7
        8 |             2 | _timescaledb_internal | _hyper_2_8_chunk  |           2 |         3
        9 |             2 | _timescaledb_internal | _hyper_2_9_chunk  |           3 |         4
       10 |             2 | _timescaledb_internal | _hyper_2_10_chunk |           4 |         5
       11 |             2 | _timescaledb_internal | _hyper_2_11_chunk |           5 |         6
       12 |             2 | _timescaledb_internal | _hyper_2_12_chunk |           6 |         7
(10 rows)

\dt "_timescaledb_internal".*
                      List of relations
        Schema         |       Name        | Type  |  Owner   
-----------------------+-------------------+-------+----------
 _timescaledb_internal | _hyper_1_2_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_3_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_4_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_5_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_6_chunk  | table | postgres
 _timescaledb_internal | _hyper_2_10_chunk | table | postgres
 _timescaledb_internal | _hyper_2_11_chunk | table | postgres
 _timescaledb_internal | _hyper_2_12_chunk | table | postgres
 _timescaledb_internal | _hyper_2_8_chunk  | table | postgres
 _timescaledb_internal | _hyper_2_9_chunk  | table | postgres
(10 rows)

SELECT _timescaledb_internal.drop_chunks_older_than(3, 'drop_chunk_test1');
NOTICE:  index "2-drop_chunk_test1_time_idx" does not exist, skipping
 drop_chunks_older_than 
------------------------
 
(1 row)

SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');
 chunk_id | hypertable_id |     chunk_schema      |    chunk_table    | range_start | range_end 
----------+---------------+-----------------------+-------------------+-------------+-----------
        3 |             1 | _timescaledb_internal | _hyper_1_3_chunk  |           3 |         4
        4 |             1 | _timescaledb_internal | _hyper_1_4_chunk  |           4 |         5
        5 |             1 | _timescaledb_internal | _hyper_1_5_chunk  |           5 |         6
        6 |             1 | _timescaledb_internal | _hyper_1_6_chunk  |           6 |         7
        8 |             2 | _timescaledb_internal | _hyper_2_8_chunk  |           2 |         3
        9 |             2 | _timescaledb_internal | _hyper_2_9_chunk  |           3 |         4
       10 |             2 | _timescaledb_internal | _hyper_2_10_chunk |           4 |         5
       11 |             2 | _timescaledb_internal | _hyper_2_11_chunk |           5 |         6
       12 |             2 | _timescaledb_internal | _hyper_2_12_chunk |           6 |         7
(9 rows)

\dt "_timescaledb_internal".*
                      List of relations
        Schema         |       Name        | Type  |  Owner   
-----------------------+-------------------+-------+----------
 _timescaledb_internal | _hyper_1_3_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_4_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_5_chunk  | table | postgres
 _timescaledb_internal | _hyper_1_6_chunk  | table | postgres
 _timescaledb_internal | _hyper_2_10_chunk | table | postgres
 _timescaledb_internal | _hyper_2_11_chunk | table | postgres
 _timescaledb_internal | _hyper_2_12_chunk | table | postgres
 _timescaledb_internal | _hyper_2_8_chunk  | table | postgres
 _timescaledb_internal | _hyper_2_9_chunk  | table | postgres
(9 rows)

